account_analytic_journal_lock_to_date/models/account_analytic_line.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2021 ForgeFlow S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime, timedelta
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT as DT
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    @api.constrains("date")
+    def _check_lock_date(self):
+        for line in self:
+            if (
+                not line.journal_id.restrict_lock_dates
+                or line.account_id.surpass_lock_dates
+            ):
+                continue
+            lock_date = line.journal_id.journal_lock_to_date
+            if not lock_date:
+                lock_date = datetime.now().date() + timedelta(days=1)
+            else:
+                lock_date = datetime.strptime(str(lock_date), DT).date()
```

Comment:
Here I think you can use lock_date directly, no need to convert

Reply:
I was getting a datetime error saying it expected string instead of datetime object

Reply:
This module is very similar and the conversion is not needed: https://github.com/OCA/account-financial-tools/blob/14.0/account_lock_to_date/models/account_move.py#L29 could you try this?

Reply:
@JasminSForgeFlow @AaronHForgeFlow I changed it, now it's working